src/EFCore.PG/Storage/Internal/NpgsqlDatabaseCreator.cs

```diff
@@ -187,6 +187,16 @@ await unpooledRelationalConnection.OpenAsync(errorsExpected: true, cancellationT
             // Pretty awful hack around #104
             return false;
         }
+        catch (NpgsqlException e) when (
+            // This can happen when Npgsql attempts to connect to multiple hosts
+            e.InnerException is AggregateException ae &&
+            ae.InnerExceptions.Any(
+                ie => ie is NpgsqlException &&
```

Comment:
@Brar are you sure there's a wrapping NpgsqlException here? I think the PostgresExceptions should just be directly beneath the AggregateException no?

Reply:
One very minor nit is also to maybe move this catch block up to be right after the first catch block dealing with IsDoesNotExist